KFSPTS-13687: Force OJB Award references to map to CuAward

cah292 opened the pull request:
This fix allows for Award-related OJB queries to map to CuAward instead, and also affects Award-related reference descriptors similarly. I did have to override the Award OJB metadata to implement this fix, but apart from adding a row-reader attribute and an extent-class element, the Award OJB XML should be the same as in base code.

This change also acts as an alternative fix to KFSPTS-13641 (from PR #665), so I have removed its related OJB XML override as well.

jhulslander commented on src/main/resources/edu/cornell/kfs/module/cg/cu-ojb-cg.xml:
```diff
@@ -14,6 +14,132 @@
+        </collection-descriptor>
+    </class-descriptor>
+    
     <class-descriptor class="edu.cornell.kfs.module.cg.businessobject.CuAward" table="CG_AWD_T">
```
Should CuAward still be mapped here?

cah292 replied:
Yes, we still need the CuAward mapping since it includes column mappings that are not in the Award OJB XML, in addition to adding a mapping for the extended attribute.

jhulslander replied:
sounds good

jhulslander commented on src/main/java/edu/cornell/kfs/sys/broker/accesslayer/RowReaderMappingBaseClassToSingleExtentClass.java:
```diff
@@ -0,0 +1,50 @@
+    protected ClassDescriptor selectClassDescriptor(Map row) throws PersistenceBrokerException {
+        ClassDescriptor baseClassDescriptor = getClassDescriptor();
+        List<?> extentClassNames = baseClassDescriptor.getExtentClassNames();
+        if (extentClassNames.size() != 1) {
```
should this be greater than 1?  I would think no extent classes defined would return super's selectClassDescriptor or perhaps baseClassDescriptor.

cah292 replied:
The extentClassNames list roughly equates to the "extent-class" elements in the OJB metadata. This RowReader is intended for very specific situations, where we want to replace queries for a base object with those of a single specific subclass (such as replacing Award with CuAward). I could modify this to be more lenient about how many (or if any) extents are defined, but that could increase the risk of overlooking configuration errors.

jhulslander replied:
Thanks for explaining, lets leave as is

jhulslander commented on src/main/java/edu/cornell/kfs/sys/broker/accesslayer/RowReaderMappingBaseClassToSingleExtentClass.java:
```diff
@@ -0,0 +1,50 @@
+        List<?> extentClassNames = baseClassDescriptor.getExtentClassNames();
+        if (extentClassNames.size() != 1) {
+            throw new PersistenceBrokerException("The class descriptor for " + baseClassDescriptor.getClassNameOfObject()
+                    + " should have had exactly 1 extent class, but instead had " + extentClassNames.size());
```
would it be worth while listing the values in extentClassNames?  If we get here, that mean there is a misconfiguration.  I am just thinking of making it easier to find the error.

cah292 replied:
Yes, that would be a good addition for troubleshooting purposes. I'll update this accordingly to print the specific extent classes.

cah292 commented:
I've made some more changes based on the initial feedback. In addition, I added some new unit tests for this functionality. The tests ended up discovering one minor issue, which I have also fixed in the latest update.

This is ready for review again.
